esp/public/media/scripts/onsite/teacher_checkin.js

```diff
@@ -108,28 +118,54 @@ $j(function(){
         this.value += "...";
     });
 
+    // Undo the check in of the user with the specified username, assuming
+    // that that user was not checked in when this page was loaded (because
+    // the DOM would look different depending on whether that were true). If
+    // no username specified, undo the last check in.
+    function undoLiveCheckIn(username) {
+        var targetCheckin;
+        if (username === undefined) {
+            if (checkins.length === 0) return false;
+            targetCheckin = checkins[checkins.length - 1];
+        } else {
+            // search from the back because we assume more recent checkins
+            // are more likely to be undone
+            for (var i = checkins.length - 1; i >= 0; i--) {
+                if (checkins[i].username === username) {
```

Comment:
I wonder if checkins should be some sort of hash rather than an array, so we don't have to traverse it every time.  But that can be a TODO for now if you want.